internal/uri/uri_test.go

```diff
@@ -1,11 +1,64 @@
-package install
+package uri
 
 import (
-	"testing"
-
 	"github.com/stretchr/testify/assert"
+	"testing"
 )
 
+func TestExtractUrl(t *testing.T) {
+	type test struct {
+		inputURL    string
+		expectedURL string
+	}
+
+	tests := []test{
+		{"https://example.com/testdb.bak,myDbName", "https://example.com/testdb.bak"},
+		{"https://example.com/testdb.bak", "https://example.com/testdb.bak"},
+		{"https://example.com,", "https://example.com,"},
+	}
+
+	for _, testcase := range tests {
+		u := NewUri(testcase.inputURL)
+		assert.Equal(t, testcase.expectedURL, u.ActualUrl(),
+			"Extracted URL does not match expected URL")
+	}
+}
+
+func TestParseDbName(t *testing.T) {
+	type test struct {
+		inputURL    string
+		expectedURL string
+	}
+
+	tests := []test{
+		{"https://example.com/testdb.bak,myDbName", "myDbName"},
+		{"https://example.com/testdb.bak", "testdb"},
+		{"https://example.com/test.foo", "test"},
+		{"https://example.com/test.foo,test", "test"},
+		{"https://example.com/test.7z,tql_name", "tsql_name"},
+		{"https://example.com/test.mdf,tsql_name?foo=bar", "tsql_name"},
```

Comment:
>"https://example.com/test.mdf,tsql_name?foo=bar [](http://example.com/codeflow?start=3&length=47)

how does putting the query portion of the URL after the comma work? It doesn't look like valid input.

A URL with a query would be `[https://example.com/test.mdf?version=10,mydatabase`](https://example.com/test.mdf?version=10,mydatabase`)